conans/model/workspace.py

```diff
@@ -1,155 +1,121 @@
         else:
-            pkg_folder = os.path.join(self.root_folder, folder)
-        mkdir(pkg_folder)
-        return pkg_folder
+            self.layout = ws_layout
 
-    @property
-    def install_folder(self):
-        return self._evaluate(self._install_folder)
-
-    @property
-    def includedirs(self):
-        return [os.path.join(self._base_folder, self._conanfile_folder, v)
-                for v in self._includedirs]
+        generators = data.get("generators")
+        if isinstance(generators, str):
+            generators = [generators]
+        if generators is None:
+            generators = ws_generators
+        self.generators = generators
 
     @property
-    def libdirs(self):
-        return [self._evaluate(v) for v in self._libdirs]
-
-    def _evaluate(self, value):
-        settings = self.conanfile.settings
-        v = value.format(build_type=settings.get_safe("build_type"),
-                         arch=settings.get_safe("arch"),
-                         os=platform.system())
-        try:
-            result = eval('%s' % v)
-        except:
-            result = v
-        return result
-
-    @property
-    def local_cmakedir(self):
-        if not self._cmakedir:
-            return self._conanfile_folder
-        return os.path.join(self._conanfile_folder, self._cmakedir).replace("\\", "/")
-
-
-WORKSPACE_FILE = "conanws.yml"
+    def root_folder(self):
+        return os.path.abspath(os.path.join(self._base_folder, self._conanfile_folder))
 
 
 class Workspace(object):
-    @staticmethod
-    def get_workspace(folder, install_folder):
-        if isinstance(folder, ConanFileReference):
-            return None
-        if not os.path.exists(folder):
-            return None
-        path = os.path.join(folder, WORKSPACE_FILE)
-        if os.path.exists(path):
-            return Workspace(path, install_folder)
-        parent = os.path.dirname(folder)
-        if parent and parent != folder:
-            return Workspace.get_workspace(parent, install_folder)
-        return None
-
-    def generate(self):
-        if self._generator == "cmake":
-            template = """# conanws
-cmake_minimum_required(VERSION 3.3)
-project({name} CXX)
-
-"""
-            cmake = template.format(name=self._name)
-            for _, workspace_package in self._workspace_packages.items():
-                build_folder = workspace_package.build_folder
-                build_folder = build_folder.replace("\\", "/")
-                cmake += 'add_subdirectory(%s "%s")\n' % (workspace_package.local_cmakedir, build_folder)
-            cmake_path = os.path.join(self._base_folder, "CMakeLists.txt")
-            if os.path.exists(cmake_path) and not load(cmake_path).startswith("# conanws"):
-                raise ConanException("Can't generate CMakeLists.txt, will overwrite existing one")
+
+    def generate(self, cwd, graph):
+        editables = {node.ref: node.conanfile for node in graph.nodes
```

Comment:
No need for this dictionary, it is only used to retrieve the `conanfile` in line 52 where you can use `node.conanfile`.

Reply:
😿

Reply:
Fixed

Reply:
😸